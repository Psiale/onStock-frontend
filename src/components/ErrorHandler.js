import React from 'react';
import Proptypes from 'prop-types';
import { useHistory } from 'react-router-dom';
import styles from './ErrorHandler.module.css';

const ErrorHandler = ({ errorMessage }) => {
  const history = useHistory();
  return (
    <div className={styles.mainContainer}>
      <h1> :C </h1>
      <h2>{errorMessage}</h2>
      <button onClick={() => history.push('/')} type="button">
        Login / Signup
      </button>
    </div>
  );
};

ErrorHandler.propTypes = {
  errorMessage: Proptypes.string.isRequired,
};

export default ErrorHandler;
