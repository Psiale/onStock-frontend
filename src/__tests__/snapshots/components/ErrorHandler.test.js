import React from 'react';
import renderer from 'react-test-renderer';
import ErrorHandler from '../../../components/ErrorHandler';

it('renders properly', () => {
  const tree = renderer.create(<ErrorHandler errorMessage="error" />).toJSON();
  expect(tree).toMatchSnapshot();
});
