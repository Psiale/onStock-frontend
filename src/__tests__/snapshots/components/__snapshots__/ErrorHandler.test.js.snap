// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders properly 1`] = `
<div
  className="mainContainer"
>
  <h1>
     :C 
  </h1>
  <h2>
    error
  </h2>
  <button
    onClick={[Function]}
    type="button"
  >
    Login / Signup
  </button>
</div>
`;
